src/__tests__/labs/ViewLab.test.tsx

```diff
@@ -66,160 +66,149 @@ describe('View Lab', () => {
       },
     } as any)
 
-    let wrapper: any
-    await act(async () => {
-      wrapper = await mount(
-        <ButtonBarProvider.ButtonBarProvider>
-          <Provider store={store}>
-            <Router history={history}>
-              <Route path="/labs/:id">
-                <titleUtil.TitleProvider>
-                  <ViewLab />
-                </titleUtil.TitleProvider>
-              </Route>
-            </Router>
-          </Provider>
-        </ButtonBarProvider.ButtonBarProvider>,
-      )
-    })
-    wrapper.find(ViewLab).props().updateTitle = jest.fn()
-    wrapper.update()
-    return { wrapper: wrapper as ReactWrapper }
+    return render(
+      <ButtonBarProvider.ButtonBarProvider>
+        <Provider store={store}>
+          <Router history={history}>
+            <Route path="/labs/:id">
+              <titleUtil.TitleProvider>
+                <ViewLab />
+              </titleUtil.TitleProvider>
+            </Route>
+          </Router>
+        </Provider>
+      </ButtonBarProvider.ButtonBarProvider>,
+    )
   }
 
-  describe('title', () => {
-    it('should have called the useUpdateTitle hook', async () => {
-      const expectedLab = { ...mockLab } as Lab
-      await setup(expectedLab, [Permissions.ViewLab])
-      expect(titleUtil.useUpdateTitle).toHaveBeenCalled()
-    })
-  })
-
   describe('page content', () => {
     it('should display the patient full name for the for', async () => {
       const expectedLab = { ...mockLab } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const forPatientDiv = wrapper.find('.for-patient')
-
-      expect(forPatientDiv.find('h4').text().trim()).toEqual('labs.lab.for')
+      setup(expectedLab, [Permissions.ViewLab])
 
-      expect(forPatientDiv.find('h5').text().trim()).toEqual(mockPatient.fullName)
+      await waitFor(() => {
+        expect(screen.getByText('labs.lab.for')).toBeInTheDocument()
+        expect(screen.getByText(mockPatient.fullName)).toBeInTheDocument()
+      })
     })
 
     it('should display the lab type for type', async () => {
       const expectedLab = { ...mockLab, type: 'expected type' } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const labTypeDiv = wrapper.find('.lab-type')
-      expect(labTypeDiv.find('h4').text().trim()).toEqual('labs.lab.type')
+      setup(expectedLab, [Permissions.ViewLab])
 
-      expect(labTypeDiv.find('h5').text().trim()).toEqual(expectedLab.type)
+      await waitFor(() => {
+        expect(screen.getByText('labs.lab.type')).toBeInTheDocument()
```

Comment:
`expect(screen.getByRole('heading', { name: /labs.lab.type/i })).toBeInTheDocument()`

more appropriate?

Reply:
I was taught getByRole is preferred over getByText. Often getByRole elements aren't available to the DOM and getByText is an easier selector to use (less code).